airflow/providers/common/sql/operators/sql.py

```diff
@@ -250,7 +250,7 @@ def execute(self, context: Context):
             records = hook.get_first(self.sql)
 
             if not records:
-                raise AirflowException(f"The following query returned zero rows: {self.sql}")
+                raise AirflowFailException(f"The following query returned zero rows: {self.sql}")
```

Comment:
I don't think this is right.
Retry can work for some senarios so we should not block it.
You can still get the behavior you wish by set retries=0

Reply:
Hello @eladkal 
Is your comment specific to this Exception or to all of them?

Let me explain why I think `retries=0` is not a good idea from my point of view. Imagine you have an infrastructure issue (By this, I mean your DB is not available due to a temporary issue, or there are connectivity issues, etc), in this particular case, you want to retry your tasks as it is an issue "external" to your query. 

Now, the change I am proposing does not affect this case, instead, it targets when the business logic is not met.  For example, we use this Operator to run simple Sanity checks (are the duplicates in the table?). The point is if you run a query and it returns False, there is no need to retry because I believe within the next retry, it will still return False. 

This is particularly useful if you run on BigQuery since retrying for queries that will always return False is a waste of money. 

WDYT?

Reply:
I'm not sure i follow?
You suggest to change the code to `AirflowFailException` which means no retries.
https://github.com/apache/airflow/blob/7efdeed5eccbf5cb709af40c8c66757e59c957ed/airflow/exceptions.py#L87-L88

>  in this particular case, you want to retry your tasks as it is an issue "external" to your query.

This is exactly why I don't believe `AirflowFailException` is right here.
`AirflowFailException` fits for cases where you must make code changes to your DAG. This is not the case here. The operator is querying database. Resolution can come from the DB side without any code changes.

Reply:
I don't understand what you mean. I think we are not understanding each other. I am speaking of something totally different. You are encapsulating when you throw this error, right? You throw this error if the condition in your SQL is not met, not if you can't reach the DB. So, the query returns True or False, then if it is False you throw the error `AirflowFailException`, in any other case you throw `AirflowException`. Can we agree on this?